src/getfields.jl

```diff
@@ -278,8 +278,16 @@ function parsefield(io::Union{IOBuffer,UnsafeBuffer}, ::Type{DateTime}, opt::CSV
                 break
             end
         end
-        val = String(pointer(io.data)+pos, position(io)-pos-quoted-1+end_of_file)
+        val = pointer_to_string(pointer(io.data)+pos, position(io)-pos-quoted-1+end_of_file)
         val == opt.null && return DateTime(0,1,1), true
         return DateTime(val, opt.dateformat)::DateTime, false
     end
 end
+
+@generated function pointer_to_string(ptr, len)
```

Comment:
Actually, I think you want `bytestring` here, which is in the latest version of Compat.